### PR TITLE
Initial implementation of the delayed TX feature.

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: src/mac_features/nrf_802154_delayed_trx.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+/**
+ * @brief Request transmission of a frame at given time.
+ *
+ * If CSMA-CA procedure is successful and frame is transmitted @sa nrf_802154_tx_started()
+ * function is called. If CSMA/CA procedure failed and frame cannot be transmitted due to busy
+ * channel @sa nrf_802154_transmit_failed() function is called.
+ *
+ * @note CSMA-CA does not timeout waiting for ACK automatically. Waiting for ACK shall be timed out
```
Comment (KaSroka) [inline src/mac_features/nrf_802154_delayed_trx.h]: What about ACK timeout feature?

File: src/mac_features/nrf_802154_delayed_trx.c [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    return result;
+}
+
+void nrf_802154_rsch_delayed_timeslot_started(void)
+{
+    bool result;
+
+    assert(tx_is_in_progress());
+
+    // TODO: Set channel and reset it in the tx_started/tx_failed hook or if the request below failed.
```
Comment (KaSroka) [inline src/mac_features/nrf_802154_delayed_trx.c]: Should this be implemented in this PR?

File: src/mac_features/nrf_802154_delayed_trx.c [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+{
+    mp_tx_psdu = NULL;
+}
+
+/**
+ * Notify MAC layer that requested timeslot is not granted if tx request failed.
+ *
+ * @param[in]  result  Result of TX request.
+ */
+static void notify_tx_timeslot_denied(bool result)
```
Comment (KaSroka) [inline src/mac_features/nrf_802154_delayed_trx.c]: This static function is not used

File: src/mac_features/nrf_802154_delayed_trx.c (hunk 2 above)
Comment (hubertmis) [reply src/mac_features/nrf_802154_delayed_trx.c]: I would add it later. It is not needed for alpha version.

File: src/mac_features/nrf_802154_delayed_trx.c (hunk 3 above)
Comment (hubertmis) [reply src/mac_features/nrf_802154_delayed_trx.c]: It is used :)

File: src/mac_features/nrf_802154_delayed_trx.h (hunk 1 above)
Comment (hubertmis) [reply src/mac_features/nrf_802154_delayed_trx.h]: Whole description here is invalid. Probably it is copy-paste. I'll update it.

File: src/mac_features/nrf_802154_delayed_trx.c (hunk 3 above)
Comment (KaSroka) [reply src/mac_features/nrf_802154_delayed_trx.c]: Oh, it's passed as a callback...

Comment (KaSroka) [reply src/mac_features/nrf_802154_delayed_trx.c]: Oh, it's passed as a callback function...

**Pull request merged**